gstd/gstd.c

```diff
@@ -221,7 +221,7 @@ main (gint argc, gchar * argv[])
     goto out;
   }
 
-  if (!nodaemon) {
+  if (daemon || kill) {
```

Comment:
Because now that daemon is not the default behavior, we must assume that if the user runs gstd -k, then gstd must be running as daemon and then we need to run the gstd_daemon_start logic to retrieve the pid to kill gstd.